traits/editor_factories.py

```diff
@@ -141,6 +142,20 @@ def date_editor():
     return DateEditor
 
 
+def datetime_editor():
+    """ Factory function that returns a Date editor with date & time for
+    editing DateTime values.
+    """
+    global DateTimeEditor
+
+    if DateTimeEditor is None:
+        from traitsui.api import DateTimeEditor
```

Comment:
Can we catch `ImportError`s here? The goal should be that we should be able to release either Traits _or_ TraitsUI first and still have the latest versions of the respective releases work with each other.